src/Application.vala

```diff
@@ -190,6 +186,31 @@ public class Mail.Application : Gtk.Application {
         main_window.present ();
     }
 
+    public async void request_background () {
+        var portal = new Xdp.Portal ();
+
+        Xdp.Parent? parent = active_window != null ? Xdp.parent_new_gtk (active_window) : null;
+
+        var command = new GenericArray<weak string> ();
+        command.add ("io.elementary.mail");
+        command.add ("--background");
+
+        try {
+            if (!yield portal.request_background (
+                parent,
+                _("Mail needs to autostart and run in background in order to send notifications even while closed."),
+                (owned) command,
+                Xdp.BackgroundFlags.AUTOSTART,
+                null
+            )) {
+                release ();
+            }
+        } catch (Error e) {
+            warning ("Failed to request autostart permissions: %s", e.message);
+            release ();
```

Comment:
i think we should let the application go to background if the portal call fail, the worse this can do is causing NotifyBackground to be called in the impl.

Reply:
For some reason the libportal implementation unfortunately throws an Error instead of returning false if the request was denied by the user. So I think we have to release here

Reply:
sorry the delay, i didn't receive a notification about this one.

looking at the libportal code, cancelling in the dialog should throw a `IOError.CANCELLED`. using the ESC key or any other type of falling throw a `IOError.FAILED`.

so we should release if we catch a `IOError.CANCELLED`, otherwise we should only show the warning.

Reply:
Sounds good, done!